ambition_utils/rrule/models.py

```diff
@@ -324,27 +328,44 @@ def generate_dates(self, num_dates=20):
         dates = []
 
         rule = self.get_rrule()
+        
+        def append_date(date):
```

Comment:
you probably don't want to name this variable `date` because it can overshadow an imported `date` module